bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/TemplatedTransformer.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2019 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.rewriter.impl;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Map;
+
+import com.adobe.acs.commons.properties.PropertyAggregatorService;
+import com.adobe.acs.commons.properties.util.TemplateReplacementUtil;
+import com.adobe.acs.commons.rewriter.ContentHandlerBasedTransformer;
+import com.adobe.granite.rest.Constants;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.rewriter.ProcessingComponentConfiguration;
+import org.apache.sling.rewriter.ProcessingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.AttributesImpl;
+
+public class TemplatedTransformer extends ContentHandlerBasedTransformer {
+
+    private static final Logger log = LoggerFactory.getLogger(TemplatedTransformer.class);
+
+    private Map<String, Object> properties;
+    private PropertyAggregatorService aggregatorService;
+
+    public TemplatedTransformer() {
+    }
+
+    public TemplatedTransformer(PropertyAggregatorService propertyAggregatorService) {
+        this.aggregatorService = propertyAggregatorService;
+    }
+
+    @Override
+    public void init(ProcessingContext processingContext, ProcessingComponentConfiguration processingComponentConfiguration) throws IOException {
+        SlingHttpServletRequest request = processingContext.getRequest();
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
+
+        // Get the combined properties via service
+        Page page = pageManager.getContainingPage(request.getResource());
+        if (page != null) {
+            properties = aggregatorService.getProperties(page);
```

Comment:
What is the runtime impact of this call to the page rendering? Can you provide basic test results on sample content? Just to get an impression of the static overhead this feature adds to the rendering of every single page (even if replacements take place or not).